manifests/server/initdb.pp

```diff
@@ -94,7 +94,7 @@
         SET encoding = pg_char_to_encoding('${encoding}'), datistemplate = TRUE
         WHERE datname = 'template1'",
       unless  => "SELECT datname FROM pg_database WHERE
-        datname = 'template1' AND pg_encoding_to_char(encoding) = '${encoding}'",
+        datname = 'template1' AND encoding = pg_char_to_encoding('${encoding}')",
```

Comment:
i'm guessing that postgresql already normalises encoding, which is why you're not passing both of those to the function, as per your comment/commit message, i.e.:
```sql
datname = 'template1' AND pg_encoding_to_char(encoding) = pg_char_to_encoding('${encoding}')
```